wasm_utils/src/api_serialization/get_entry.rs

```diff
@@ -72,3 +87,82 @@ impl EntryHistory {
         }
     }
 }
+
+#[derive(Deserialize, Debug, Serialize, DefaultJson)]
+pub struct GetEntryResult {
+    pub found: bool, // indicates search was successful
+    // pub header: Option<ChainHeader>,   // header if requested in options
+    // pub sources: Option<Vec<Address>>, // sources if requested in options
+    pub history: Option<EntryHistory>,
+}
+impl GetEntryResult {
+    pub fn new() -> Self {
+        GetEntryResult {
+            found: false,
+            //            sources: None,
+            history: None,
+        }
+    }
+    pub fn found(&self) -> bool {
+        self.found
+    }
+
+    fn add_history(&mut self) {
+        self.history = Some(EntryHistory::new())
+    }
+
+    pub fn push(&mut self, entry_with_meta: &EntryWithMeta) {
+        if !self.found {
+            self.found = true;
+            self.add_history();
+        }
+        match self.history {
+            None => unreachable!(),
+            Some(ref mut history) => history.push(entry_with_meta),
+        };
+    }
+
+    pub fn latest(&self) -> Option<Entry> {
+        if !self.found() {
+            return None;
+        }
+        let entry = match self.history {
+            None => None,
+            Some(ref h) => Some(h.entries.iter().next().unwrap().clone()),
+        };
+        entry
```

Comment:
You can rewrite this and make it much cleaner:
```rust
self.history
    .filter(|_| self.found())
    .map(|h| h.entries[0].clone())
```

Reply:
I see this pattern of using `match` to work with an `Option<T>` way too often. Most of the matches you can and want to do on an `Option<T>` can be done with what's in the standard library: https://doc.rust-lang.org/std/option/enum.Option.html
I invite you to take a look at all those beautiful functions 😄

Reply:
Sweet.  fixed and then changed.